web/courses.go

```diff
@@ -533,6 +539,61 @@ func (s *AutograderService) setLastApprovedAssignment(submission *pb.Submission,
 	return s.db.UpdateEnrollment(query)
 }
 
+// acceptRepositoryInvites tries to accept repository invitations for the given repository types on behalf of the given user.
+func (s *AutograderService) acceptRepositoryInvites(ctx context.Context, sc scm.SCM, user *pb.User, course *pb.Course, repoTypes []pb.Repository_Type) {
+	invites := []*github.RepositoryInvitation{}
+	for _, repoType := range repoTypes {
+		repo, err := s.db.GetRepositoryByType(user.ID, course.OrganizationID, repoType)
+		if err != nil {
+			s.logger.Error("Failed to get repo: %s", err)
+			continue
+		}
+		// TODO: pb.Repository should contain Path
+		path := strings.Split(repo.GetHTMLURL(), "/")
+
+		repoInvites, err := sc.GetRepositoryInvites(ctx, &scm.RepositoryOptions{ID: repo.RepositoryID, Owner: course.OrganizationPath, Path: path[len(path)-1]})
+		if err != nil {
+			s.logger.Errorf("Failed to get repository invites for %v: %s - %s", repo.RepoType, user.Login, err)
+			continue
+		}
+		for _, invite := range repoInvites {
+			// Some repositories contain more than one invite
+			// and we only want to accept the invite for the
+			// user we are enrolling.
+			if *invite.Invitee.Login != user.Login {
+				continue
+			}
+			invites = append(invites, invite)
+		}
+	}
+
+	if !(len(invites) > 0) {
+		return
+	}
+
+	user, err := s.db.GetUser(user.ID)
```

Comment:
The `pb.User` object is already passed in as an argument to the `acceptRepositoryInvites` method; so this database lookup is unnecessary.

Reply:
The user we get here:
```go
course, user := enrolled.GetCourse(), enrolled.GetUser()
```
does not come preloaded with `RemoteIdentity` (from `GetEnrollmentByCourseAndUser`). We could preload `RemoteIdentity` to avoid this lookup, but I need to check if that would have some unintended side-effects.

Reply:
That's strange because the code does seem to preload it:
```go
func (db *GormDB) GetUser(userID uint64) (*pb.User, error) {
	var user pb.User
	if err := db.conn.Preload("RemoteIdentities").First(&user, userID).Error; err != nil {
		return nil, err
	}
	return &user, nil
}
```

Reply:
Looks like the new solution to use `s.getSCM()` for the user reasonable. Although I wish we didn't have to specify the provider here... But that's another issue.